### PR TITLE
Update study.py for sphinx

### PR DESCRIPTION
**Pull request by eukaryo**
## Motivation

Fix [issue#4917](https://github.com/optuna/optuna/issues/4917).

`sphinx` could not recognize `pd.DataFrame`, which is the return type of `trials_dataframe` function, and exerts an warning in `nitpicky` mode.

## Description of the changes

I used `sys.modules` in order to check whether `sphinx` was imported,. If so, let it to import`pd`.

### CODE REVIEW

File: optuna/study/study.py [hunk 1]
```diff
@@ -42,7 +43,7 @@
 
 _dataframe = _LazyImport("optuna.study._dataframe")
 
-if TYPE_CHECKING:
+if TYPE_CHECKING or "sphinx" in sys.modules:
     from optuna.study._dataframe import pd
```
Comment (toshihikoyanase) [inline optuna/study/study.py:47]: I'm not fully sure, but if users have Sphinx installed without Pandas, they might encounter an import error. Could you please look into this?

Comment (eukaryo) [reply optuna/study/study.py:47]: When I uninstalled `pandas` and then performed `made html`, I got a warning from `study.py`, but `plotly` gave import errors regardless of this PR.
However, 
```
>>> import optuna
>>> import sys
>>> 'sphinx' in sys.modules
False
>>> import sphinx
>>> 'sphinx' in sys.modules
True
```
It seems to be not a good behavior.

Comment (eukaryo) [reply optuna/study/study.py:47]: On second thought, I thought it would not be a problem. I think lazy-import will prevent import error.

Comment (toshihikoyanase) [reply optuna/study/study.py:47]: Thank you for your investigation.
BTW, this PR seems to be still work-in-progress, so let me remove the assignment for now.

**Pull request closed without merging**